File: read_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

let dataBuffer = fs.readFileSync('hu_kegyes_Koran_magyar.pdf');

pdf(dataBuffer).then(function(data) {
    // number of pages
    console.log("Num Pages:", data.numpages);
    // PDF info
    console.log("Info:", data.info);
    
    const lines = data.text.split('\n');
    console.log("Total lines:", lines.length);
    
    // log first 500 lines to see the structure
    console.log("First 500 lines:");
    for(let i=0; i<500; i++) {
        if(lines[i]) console.log(lines[i]);
    }
}).catch(function(error){
    console.log("Error:", error);
});

File: read_pdf.py
import fitz  # PyMuPDF

def read_pdf(file_path):
    doc = fitz.open(file_path)
    
    # Check random pages
    for i in [20, 50, 100, 200, 300, 400]:
        page = doc.load_page(i)
        text = page.get_text()
        print(f"--- Page {i} ---")
        lines = [line.strip() for line in text.split('\n') if line.strip()]
        if lines:
            for line in lines[:30]:
                print(line)
        else:
            print("No text found.")

if __name__ == "__main__":
    read_pdf("hu_kegyes_Koran_magyar.pdf")
